backend/compat.py

```diff
@@ -1 +1,13 @@
+# Werk van Sil
 
+# for-loop
+# [] 
+
+
+def validate_appointment(data: dict):
+    KeyList = ['title', 'description', ...]
+    for key in KeyList:
```

Comment:
> X stelt één key uit de dictionary voor en y is die list. Dus zolang er nog een key in de dictionary zit die de loop nog niet gehad heeft gaat de loop door.

Voor het geval van de dictionary klopt dit inderdaad. Bekijk het echter eens algemener: Een for loop kan over veel meer verschillende dingen loopen, bijvoorbeeld ook een `list`

> Range genereerd nummers en daardoor kun je hem gebruiken voor loops.

Ja. Wat is er speciaal aan de manier waarop range nummers genereert?

> Enumerate geeft elke variabele uit een list een waarde beginnend bij 0.

Kun je een voorbeeld geven van hoe je enumerate zou gebruiken in een voorbeeld?

> hij print de keys namelijk 1 en 3, de 2 en 4 zijn de values.

Klopt

> 'for value in {1: 2, 3: 4}: print(value)'

Dit klopt niet. `for key in dict` loopt altijd over de keys, ongeacht de naam van de variabele in de for-loop.

> Hij print de waardes 1 2 3 en 4, deze items zitten in een list en hiervoor zaten de keys met values in een dictionary.(zit dus verschil tussen)

close enough